saleor/core/taxes/interface.py

```diff
@@ -0,0 +1,179 @@
+from .vatlayer import interface as vatlayer_interface
+
+if TYPE_CHECKING:
+    from ...checkout.models import Checkout, CheckoutLine
+    from ...product.models import Product
+    from ...account.models import Address
+    from ...order.models import OrderLine, Order
+
+
+def calculate_checkout_total(
+    checkout: "Checkout", discounts: List["DiscountInfo"]
+) -> TaxedMoney:
+    """Calculate total gross for checkout"""
+
+    if settings.VATLAYER_ACCESS_KEY:
+        return vatlayer_interface.calculate_checkout_total(checkout, discounts)
+    elif settings.AVATAX_USERNAME_OR_ACCOUNT and settings.AVATAX_PASSWORD_OR_LICENSE:
+        return avatax_interface.calculate_checkout_total(checkout, discounts)
+
+    total = checkout.get_total(discounts)
+    return TaxedMoney(net=total, gross=total)
+
+
+def calculate_checkout_subtotal(
+    checkout: "Checkout", discounts: List["DiscountInfo"]
+) -> TaxedMoney:
+    """Calculate subtotal gross for checkout"""
+
+    if settings.VATLAYER_ACCESS_KEY:
+        return vatlayer_interface.calculate_checkout_subtotal(checkout, discounts)
+    elif settings.AVATAX_USERNAME_OR_ACCOUNT and settings.AVATAX_PASSWORD_OR_LICENSE:
+        return avatax_interface.calculate_checkout_subtotal(checkout, discounts)
+
+    subtotal = checkout.get_subtotal(discounts)
+    return TaxedMoney(net=subtotal, gross=subtotal)
+
+
+def calculate_checkout_shipping(
+    checkout: "Checkout", discounts: List["DiscountInfo"]
+) -> TaxedMoney:
+    """Calculate shipping gross for checkout"""
+    if not checkout.shipping_method:
+        return ZERO_TAXED_MONEY
+    if settings.VATLAYER_ACCESS_KEY:
+        return vatlayer_interface.calculate_checkout_shipping(checkout, discounts)
+    elif settings.AVATAX_USERNAME_OR_ACCOUNT and settings.AVATAX_PASSWORD_OR_LICENSE:
+        return avatax_interface.calculate_checkout_shipping(checkout, discounts)
+    total = checkout.get_shipping_price()
+    return TaxedMoney(net=total, gross=total)
+
+
+def calculate_order_shipping(order: "Order") -> TaxedMoney:
+    """Calculate shipping price that assigned to order"""
+    if settings.VATLAYER_ACCESS_KEY:
+        return vatlayer_interface.calculate_order_shipping(order)
+    elif settings.AVATAX_USERNAME_OR_ACCOUNT and settings.AVATAX_PASSWORD_OR_LICENSE:
+        return avatax_interface.calculate_order_shipping(order)
+    price = order.shipping_method.price
+    return TaxedMoney(net=price, gross=price)
+
+
+def apply_taxes_to_shipping(price: Money, shipping_address: "Address") -> TaxedMoney:
+    """Apply taxes for shipping methods that user can use during checkout"""
+    if shipping_address:
+        if settings.VATLAYER_ACCESS_KEY:
+            return vatlayer_interface.apply_taxes_to_shipping(price, shipping_address)
+        if settings.AVATAX_USERNAME_OR_ACCOUNT and settings.AVATAX_PASSWORD_OR_LICENSE:
+            # For now we can't calculate product/shipping prices that are not directly
```

Comment:
Might want to throw a warning?

Reply:
I am not sure, this will populate a huge amount of warning. I assume that this is expected because avalara doesn't have any way to calculate prices without shipping address.

Reply:
Fair enough. As long as it gets documented.